### PR TITLE
GraphPanel: Don't sort series when legend table & sort column is not visible 

### LINKED ISSUE #16980: Graph: Only sort series in legend table mode
**What happened**:
sortByName() appears to be broken in Grafana 6.1.6

**What you expected to happen**:
We run a query to get back host, filename metrics.  We do an aliasByTags(host, fileSystem) and then sortByName(true, false)

We can use the query inspector to confirm that the data is properly sorted in the result set, however the legend in Grafana is not sorted.

**How to reproduce it (as minimally and precisely as possible)**:

Here is our MT query:
sortByName(aliasByTags(consolidateBy(seriesByTag('namespace=os', 'name=fileSystem.bytes.used.g', 'cluster=~${my_cluster:regex}', 'fileSystem=$my_fileSystem'), 'max'), 'cluster', 'host', 'fileSystem'), true, false)

**Anything else we need to know?**:

**Environment**:
- Grafana version: 6.1.6
- Data source type & version: MetricTank
- OS Grafana is installed on: Linux
- User OS & Browser: Windows, Chrom
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by shavonn**
Fixes #16980

### CODE REVIEW

**Comment by torkelo** [comment]
I am not sure this fixes the problem described in the issue. Hold off merging, need to investigate

**Review by torkelo — CHANGES_REQUESTED** [review]
This is not really fixing the problem fully and has an update call (onToggleSort) in the render path. 
Think ignoring the sort option if alignAsTable is false or legend[legend.sort] is false is best. 

There are two sorts happening (took me some time to figure out why) but think it's correct as they can be independent. 

So the logic of not sorting if alignAsTable is false or legend[legend.sort] is false needs to be done here as well. 
https://github.com/grafana/grafana/blob/master/public/app/plugins/panel/graph/graph.ts#L461

The reason there are 2 sort is that one controls the sort in legend, the other sort in the graph (render order), the sort in the graph only has effect if stacking is enabled.

File: public/app/plugins/panel/graph/Legend/Legend.tsx [hunk 1]
```diff
@@ -100,6 +100,15 @@ export class GraphLegend extends PureComponent<GraphLegendProps, LegendState> {
       if (this.props.sortDesc) {
         seriesList = seriesList.reverse();
       }
+    } else {
+      const { min, max, avg, current, total, values } = this.props;
+      const seriesValuesProps = { min, max, avg, current, total, values };
+      for (const property in seriesValuesProps) {
+        if (seriesValuesProps[property]) {
+          this.props.onToggleSort(property, this.props.sortDesc);
```
[the remark this answers is not in the document]
Comment (shavonn) [reply public/app/plugins/panel/graph/Legend/Legend.tsx]: Removed this entire block in lieu of proposed solution above.

**Comment by daniellee** [comment]
@shavonn after the last changes, there is a failing test: https://circleci.com/gh/grafana/grafana/100809?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link

**Review by torkelo — APPROVED** [review]
Looks good now!

**Pull request merged**